src/Nethermind/Nethermind.AuRa.Test/Transactions/TxPermissionFilterTest.cs

```diff
@@ -257,7 +257,7 @@ protected override BlockProcessor CreateBlockProcessor()
                 TransactionPermissionContractVersions =
                     new LruCache<Keccak, UInt256>(PermissionBasedTxFilter.Cache.MaxCacheSize, nameof(TransactionPermissionContract));
 
-                var trieStore = new ReadOnlyTrieStore(new TrieStore(DbProvider.StateDb, LimboLogs.Instance));
+                var trieStore = new TrieStore(DbProvider.StateDb, LimboLogs.Instance).AsReadOnly(DbProvider.StateDb);
```

Comment:
AsReadOnly() would be better here

Reply:
Would only work on TrieStore or we need to put it into the interface.